tests/providers/google/cloud/utils/test_credentials_provider.py

```diff
@@ -349,7 +367,7 @@ def test_disable_logging(self, mock_default, mock_info, mock_file):
                 key_path='KEY.json',
                 disable_logging=True,
             )
-        assert logs.output == ['DEBUG:root:nothing']
+        assert_no_logs(logs.records)
```

Comment:
Hey @Taragolis , building off your work, i got inspired, and figured out we could wrap this logic in a context mgr and make it all even cleaner.

No obligation to accept it but here is the code

```suggestion
        assert_no_logs(logs.records)

    @contextmanager
    def assert_no_logs(self, name, level):
        with self.assertLogs(level=level) as logs:
            logging.debug('nothing')
            yield
        records = [log_record for log_record in logs.records if log_record.name == name]
        if not records:
            return
        raise AssertionError(f"Did not expect any log message from logger={name!r}, but got: {records}")


```

Reply:
then the code becomes this:
```python
        # assert no logs
        with self.assert_no_logs(name=logger_name, level="DEBUG") as logs:
            get_credentials_and_project_id(disable_logging=True)

        # assert no debug logs emitted from get_credentials_and_project_id
        with self.assert_no_logs(name=logger_name, level="DEBUG") as logs:
            get_credentials_and_project_id(
                keyfile_dict={'private_key': 'PRIVATE_KEY'},
                disable_logging=True,
            )

        # assert no debug logs emitted from get_credentials_and_project_id
        with self.assert_no_logs(name=logger_name, level="DEBUG") as logs:
            get_credentials_and_project_id(
                key_path='KEY.json',
                disable_logging=True,
            )

```